packages/core/src/node-execution-context/supply-data-context.ts

```diff
@@ -173,6 +172,11 @@ export class SupplyDataContext extends BaseExecuteContext implements ISupplyData
 			);
 		});
 
+		if (!Array.isArray(this.inputData[connectionType])) {
+			this.inputData[connectionType] = [];
+		}
+		this.inputData[connectionType].push(...data);
```

Comment:
this is wrong, as it will now flatten responses from multiple invocations of the tool into a single array.